### ISSUE TITLE
Features are always disabled

### ISSUE THREAD

**Issue by anogr** [Unleash/unleash#2782]
I'm new to Unleash. I have installed the self-hosted version and got it up and running on a K8s cluster. Everything appears fine.

So I created a feature toggle, and enabled it in development and production. Then, I created a Node.js app as described in the Node.js SDK docs. I configured the API key, set the environment to development and the call to startUnleash() appears to be fine.

Then, if I check if the feature toggle is enabled, it returns false. I can see that the feature toggle was found because the fallback function was not called. I tried with a feature toggle that isn't defined and the fallback function was called so Unleash actually knows that the feature toggle exists. Strange.

Even more strange, if I make an explicit HTTP call to my.server.host/api/client/features I get the feature toggle in the response and can see that enabled is false. The environments, development and production, are not listed so it's unclear what the value actually means. Why isn't the environments included in the response?

If I make a call to my.server.host/api/admin/features I get a similar response with the same problem. No environments in the response and the wrong "enabled" value for the feature toggle.

If I make a call to my.server.host/api/admin/projects/default I can see a correct response with both environments and correct values for enabled.

In summary, the Node.js SDK and two out of three APIs appear to return the wrong value for the feature toggle. Why is that? It appears to be some bug in the two APIs and the Node.js SDK appears to be using the failing APIs instead of the one that actually works.

**Comment by anogr** [comment]
I found the error. I must use an API key that's targeted for clients with a specific environment, and I need to provide the environment as a query parameter for the API calls to respond with correct values.

Well, if you read this, a little more helpful responses or error messages for stupid people like me wouldn't be a bad thing.

**Comment by thomasheartman** [comment]
Hey, @anogr! Thanks for the issue and sorry you found it confusing. I think you've partially found the solution, but not quite:

When you use an [client API token](https://docs.getunleash.io/reference/api-tokens-and-client-keys#client-tokens), the environment is encoded into the token itself, and you don't need to specify it as a query parameter. The `environment` config option is **not used to set the environment**, but only left in for compatibility with older versions  of Unleash. 

So if you [create an API token (server-side)](https://docs.getunleash.io/how-to/how-to-create-api-tokens) for the desired environment, that should be enough.

That said, looking through the node docs, I see that it's a bit out of date. I'll see if I can't make some small changes there to make it more obvious. Thanks for letting us know 😄

**Comment by kjvalencik** [comment]
This doesn't appear to work as expected. When `AuthType.NONE` is set, the `apiTokenMiddleware` doesn't get added. Without the middleware, the `environment` isn't parsed from the token.

The middleware should be loaded as is--to require API tokens even without UI auth--or it should be wrapped to ignore errors so that it works with or without.

https://github.com/Unleash/unleash/blob/627958d30f01a8124c6f25c76efb16befd815d2c/src/lib/app.ts#L138
